### PR TITLE
many: introduce changelog for the snap repository itself prior to package changes

### PR DESCRIPTION
**Pull request by Meulengracht**
Example of changelog with this change: https://paste.ubuntu.com/p/qcyYNVwMXz/

### CODE REVIEW

**Comment by Meulengracht** [comment]
I need to fix the CI build which throws an error :-)

File: tools/generate-changelog.py [hunk 1]
```diff
@@ -182,30 +196,47 @@ def read_remote_git_url() -> str:
...
     now = datetime.now()
-    changes = f"{now.strftime("%d/%m/%Y")}, commit {read_remote_git_url()}/tree/{read_commit_hash()}\n\n"
+    changes = f"{now.strftime("%d/%m/%Y")}, commit {read_remote_git_url()}/tree/{ccommit}\n\n"
+    changes += f'[ Changes in the {args.name} snap ]\n\n'
+
+    if pcommit != "":
+        changes += log_between_commits(pcommit, ccommit)
+        changes += "\n\n"
+    else:
+        changes += f'No changes for the core{args.name} snap\n\n'
```
Comment (alfonsosanchezbeato) [inline tools/generate-changelog.py:231]: But if pcommit==ccommit we would not have changes either, right? In fact if pcommit=="" what might be happening is that there is no previous changelog?

File: tools/generate-changelog.py [hunk 2]
```diff
@@ -182,30 +196,47 @@ def read_remote_git_url() -> str:
     return remote_url.removesuffix(".git")
 
 
+def log_between_commits(start, end):
+    return subprocess.check_output(['git', 'log', f'{start}..{end}']).decode()
```
Comment (alfonsosanchezbeato) [inline tools/generate-changelog.py:200]: I think we should take only the title of the commit, having the full commit information is too noisy I think

**Review by alfonsosanchezbeato — COMMENTED** [review]
Thanks! Couple of comments.

File: tools/generate-changelog.py (hunk 2 above)
Comment (Meulengracht) [reply tools/generate-changelog.py:200]: How about we use `git shortlog` instead?

File: tools/generate-changelog.py (hunk 1 above)
Comment (Meulengracht) [reply tools/generate-changelog.py:231]: Yea that is a nice catch, thank you, i'll just test some more cases here

File: tools/generate-changelog.py (hunk 2 above)
Comment (Meulengracht) [reply tools/generate-changelog.py]: Now it looks like this: https://paste.ubuntu.com/p/qcyYNVwMXz/

Comment (Meulengracht) [reply tools/generate-changelog.py]: Agreed, something we need to be aware of when merging, that commits look nice

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks for the changes

**Pull request merged**